src/Microsoft.ServiceFabric.Services.Remoting/Description/MethodDescription.cs

```diff
@@ -61,10 +61,11 @@ public MethodInfo MethodInfo
      
         internal static MethodDescription Create(string remotedInterfaceKindName, MethodInfo methodInfo)
         {
-            var argumentList = new List<MethodArgumentDescription>();
+            var parameters = methodInfo.GetParameters();
+            var argumentList = new List<MethodArgumentDescription>(parameters.Length);
             var hasCancellationToken = false;
 
-            foreach (var param in methodInfo.GetParameters())
+            foreach (var param in parameters)
```

Comment:
My comment above about list growing was referencing this code change. In this example, we know argumentList will either be parameters.Length or parameters.Length-1. Since we know the maximum size, we can create our list with capacity of parameters.Length to avoid resizing from 0->4->8->16->etc.
The important change here is:
var argumentList = new List<MethodArgumentDescription>(**parameters.Length**);